astroquery/astrometry_net/core.py

```diff
@@ -0,0 +1,388 @@
+                               ):
+        """
+        Plate solve from a list of source positions.
+
+        Parameters
+        ----------
+
+        x : list-like
+            List of x-coordinate of source positions.
+        y : list-like
+            List of y-coordinate of source positions.
+        image_width : int
+            Size of the image in the x-direction.
+        image_height : int
+            Size of the image in the y-direction.
+        """
+        settings = {
+            # 'allow_commercial_use': allow_commercial_use,
+            # 'allow_modifications': allow_modifications,
+            # 'publicly_visible': publicly_visible,
+            'scale_units': scale_units,
+            'scale_type': scale_type,
+            'scale_lower': scale_lower,
+            'scale_upper': scale_upper,
+            'scale_est': scale_est,
+            'scale_err': scale_err,
+            'center_ra': center_ra,
+            'center_dec': center_dec,
+            'radius': radius,
+            # 'downsample_factor': downsample_factor,
+            'tweak_order': tweak_order,
+            # 'use_sextractor': use_sextractor,
+            'crpix_center': crpix_center,
+            'parity': parity,
+            'image_width': image_width,
+            'image_height': image_height,
+            # 'positional_error': positional_error,
+        }
+        if (x is None or y is None or
+                image_width is None or image_height is None):
+            raise ValueError('Must provide values for x, y, '
+                             'image_width and image_height')
+        settings = {k: v for k, v in six.iteritems(settings) if v is not None}
+        settings['x'] = [float(v) for v in x]
+        settings['y'] = [float(v) for v in y]
+        self._validate_settings(settings)
+        if self._session_id is None:
+            self._login()
+
+        settings['session'] = self._session_id
+        payload = self._contruct_payload(settings)
+        url = url_helpers.join(self.API_URL, 'url_upload')
+        response = self._request('POST', url, data=payload, cache=False)
+        if response.status_code != 200:
+            raise RuntimeError('Post of job failed')
+        response_d = response.json()
+        submission_id = response_d['subid']
+        has_completed = False
+        job_id = None
+        while not has_completed:
+            time.sleep(1)
+            sub_stat_url = url_helpers.join(self.API_URL, 'submissions', str(submission_id))
+            sub_stat = self._request('GET', sub_stat_url, cache=False)
+            jobs = sub_stat.json()['jobs']
+            if jobs:
+                job_id = jobs[0]
+            if job_id:
+                job_stat_url = url_helpers.join(self.API_URL, 'jobs', str(job_id), 'info')
+                job_stat = self._request('GET', job_stat_url, cache=False)
+                has_completed = job_stat.json()['status'] == 'success'
+            print(has_completed)
```

Comment:
maybe `print('.', end='')` unless printing success?

Reply:
This was a leftover from debugging; however, should I have a progress indicator as you suggested? When I was running test solves Wednesday night it often took 30-40 seconds to complete the solution, so some indication that something is happening might be useful for interactive use.